core/src/main/java/org/everit/json/schema/ReferenceSchema.java

```diff
@@ -145,7 +145,6 @@ public boolean equals(Object o) {
             return that.canEqual(this) &&
                     Objects.equals(refValue, that.refValue) &&
                     Objects.equals(unprocessedProperties, that.unprocessedProperties) &&
-                    Objects.equals(referredSchema, that.referredSchema) &&
```

Comment:
Hello @tmarsteel , do you think we loose anything if we keep this line? So my concern is that, with this change, there can be (rare) cases when two schemas are considered equal when they are not (I mean eg. two references pointing to `#` , but this denotes the roots of two different schema documents). 
What do you think about keeping this line? I see that it might run into infinite `equals()` recursions, but that's probably a very rare case.

Reply:
In the case of Confluent Schema Registry, we call `equals` quite often on schema.   So we would prefer the fix above.

Reply:
That's a very good point. The testcase and my actual cyclic schemas both work with `referredSchema` being included in equals.